### PR TITLE
Remove ResultPartial

### PR DESCRIPTION
**Pull request by breeswish**
## What have you changed? (mandatory)

Our current aggregation implementation (as well as HashAgg in future) cannot utilize `ResultPartial` well, which doesn't improve performance compared to `[VectorValue]`. So let's simply remove `ResultPartial`.

- [x] Merge https://github.com/tikv/tikv/pull/4726 first.

## What are the type of the changes? (mandatory)

- Engineering (engineering change which doesn't change any feature or fix any issue)

## How has this PR been tested? (mandatory)

Existing tests.

### CODE REVIEW

File: src/coprocessor/dag/aggr_fn/impl_count.rs [hunk 1]
```diff
@@ -119,26 +119,14 @@ impl<T: Evaluable> super::AggrFunctionStateUpdatePartial<T> for AggrFnStateCount
...
-        panic!("Unmatched result append target type")
-    }
-}
-
-impl super::AggrFunctionStateResultPartial<Vec<Option<Int>>> for AggrFnStateCount {
-    #[inline]
-    fn push_result(&self, _ctx: &mut EvalContext, target: &mut Vec<Option<Int>>) -> Result<()> {
-        target.push(Some(self.count as Int));
+    fn push_result(&self, _ctx: &mut EvalContext, target: &mut [VectorValue]) -> Result<()> {
+        target[0].push(Some(self.count as Int));
```
Comment (lonng) [inline src/coprocessor/dag/aggr_fn/impl_count.rs]: Is necessary to add an assert to here, like `impl_avg:L159`?

Comment (breeswish) [reply src/coprocessor/dag/aggr_fn/impl_count.rs]: Maybe. I can add it in future.

**Pull request merged**